pkg/util/kubernetes/apiserver/apiserver.go

```diff
@@ -92,10 +94,50 @@ func (c *APIClient) connect() error {
 	ctx, cancel := context.WithTimeout(context.Background(), c.timeout)
 	defer cancel()
 	version, err := c.client.Discovery().Version(ctx)
-	if err == nil {
-		log.Debugf("connected to kubernetes apiserver, version %s", version.GitVersion)
+	if err != nil {
+		return err
+	}
+	log.Debugf("connected to kubernetes apiserver, version %s", version.GitVersion)
+
+	resourcesAuthProbError := c.resourcesAuthProb()
+	if resourcesAuthProbError != nil {
+		return resourcesAuthProbError
+	}
+	log.Debug("Could successfully collect Pods, Nodes, Services and Events.")
+	return nil
+}
+
+func (c *APIClient) resourcesAuthProb() error {
+	ctx, cancel := context.WithTimeout(context.Background(), c.timeout)
+	defer cancel()
+	var resourcesError error
+	var serviceMapperAuthError string
+	var eventAuthError string
+
+	_, errEvents := c.client.CoreV1().ListEvents(ctx, "")
+	if errEvents != nil {
+		eventAuthError = fmt.Sprintf("Event collection error: %s", errEvents.Error)
+	}
+	_, errServices := c.client.CoreV1().ListServices(ctx, "")
+	if errServices != nil {
+		serviceMapperAuthError = fmt.Sprintf("Service collection error: %s", errServices.Error)
+	}
+	_, errPods := c.client.CoreV1().ListPods(ctx, "")
+	if errPods != nil {
+		serviceMapperAuthError = fmt.Sprintf("%s \nPods collection error: %s", serviceMapperAuthError, errPods.Error)
+	}
+	_, errNodes := c.client.CoreV1().ListNodes(ctx)
+	if errNodes != nil {
+		serviceMapperAuthError = fmt.Sprintf("%s \nNodes collection error: %s", serviceMapperAuthError, errNodes.Error)
+	}
+
+	if eventAuthError != "" && config.Datadog.GetBool("collect_events") {
+		resourcesError = errors.New(fmt.Sprintf("cannot collect events: %s /n", eventAuthError))
+	}
+	if serviceMapperAuthError != "" && config.Datadog.GetBool("use_service_mapper") {
+		resourcesError = errors.New(fmt.Sprintf("%s cannot compute the service mapper: %s", resourcesError, serviceMapperAuthError))
```

Comment:
If we concatenate the previous error, it ends with `\n` so you can remove the white space between the `%s` and `cannot`:
`%scannot compute...`